solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/Feature.java

```diff
@@ -34,13 +34,15 @@
 public abstract class Feature extends Query implements Cloneable {
 
   protected String name;
+  protected String type;
   protected Normalizer norm = IdentityNormalizer.INSTANCE;
   protected int id;
   protected NamedParams params = NamedParams.EMPTY;
 
   public void init(String name, NamedParams params, int id)
       throws FeatureException {
     this.name = name;
+    this.type = getClass().getSimpleName();
```

Comment:
Is it necessary for us to store this locally like this? The concern I have is that at present at least toString seems to be only called when explain is enabled (let me know if I'm wrong). Meanwhile there could be many thousands of features in some cases and now each of them needs to store a number of bytes for "type", which seems potentially inefficient.